test_handler.go

```diff
@@ -73,10 +82,32 @@ func testHandler(w http.ResponseWriter, r *http.Request) {
         http.Error(w, err.Error(), http.StatusInternalServerError)
         return
     }
-    testRunsJSON := string(testRunsBytes)
 
-    if err := templates.ExecuteTemplate(w, "index.html", testRunsJSON); err != nil {
+    data := struct {
+        TestRuns string
+        SHA      string
+    }{
+        string(testRunsBytes),
+        runSHA,
+    }
+
+    if err := templates.ExecuteTemplate(w, "index.html", data); err != nil {
         http.Error(w, err.Error(), http.StatusInternalServerError)
         return
     }
 }
+
+func getRunSHA(r *http.Request) (runSHA string, err error) {
+    // Get the SHA for the run being loaded (the first part of the path.)
+    runSHA = "latest"
+    params, err := url.ParseQuery(r.URL.RawQuery)
+    if err != nil {
+        return
+    }
+
+    runParam := params.Get("run")
+    if runParam != "" && len(runParam) == 10 {
```

Comment:
Idea: how about something like

```go
import "regexp"

if !regexp.MatchString("[a-z0-9]{10}", runParam) {
```

(I haven't compiled that though)

Reply:
Done.